lib/systems/examples.nix

```
@@ -221,6 +221,28 @@ rec {
     libc = "newlib";
   };
 
+  #
+  # UEFI
+  #
+
+  i686-uefi = {
+    config = "i686-unknown-windows-gnu";
+    rust.config = "x86_64-unknown-uefi";
+    useLLVM = true;
+  };
+
+  x86_64-uefi = {
+    config = "x86_64-unknown-windows";
+    rust.config = "x86_64-unknown-uefi";
+    useLLVM = true;
+  };
+
+  aarch64-uefi = {
+    config = "aarch64-unknown-windows";
+    rust.config = "aarch64-unknown-uefi";
+    useLLVM = true;
+  };
```

Currently these two configs don't make it through config parsing (triples with "windows" as the OS and the ABI implied aren't handled by `lib/systems/parse.nix`).

If I'm understanding [this part of `Triple.cpp`](https://github.com/llvm/llvm-project/blob/293e4da32b1d823e63b2614e626bcd22649a8a15/llvm/lib/TargetParser/Triple.cpp#L1121-L1129) correctly, ABI gets set to `msvc` for these triples.

We can either update the logic in `parse.nix` ([here](https://github.com/NixOS/nixpkgs/blob/f0a2e06ad29cac466635d746178c11b828e50224/lib/systems/parse.nix#L427-L428) I think) to support triples of this form or we can just tack on `-msvc` to the configs here:

```suggestion
  x86_64-uefi = {
    config = "x86_64-unknown-windows-msvc";
    rust.config = "x86_64-unknown-uefi";
    useLLVM = true;
  };

  aarch64-uefi = {
    config = "aarch64-unknown-windows-msvc";
    rust.config = "aarch64-unknown-uefi";
    useLLVM = true;
  };
```

I'm not confident I'm not misinterpreting `Triple.cpp` though; others that are familiar with this should definitely weigh in.